packages/contracts-bedrock/test/kontrol/proofs/utils/KontrolUtils.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Vm } from "forge-std/Vm.sol";
+import { Types } from "src/libraries/Types.sol";
+import { KontrolCheats } from "kontrol-cheatcodes/KontrolCheats.sol";
+
+contract GhostBytes {
+    bytes public ghostBytes;
+}
+
+contract GhostBytes10 {
+    bytes public ghostBytes0;
+    bytes public ghostBytes1;
+    bytes public ghostBytes2;
+    bytes public ghostBytes3;
+    bytes public ghostBytes4;
+    bytes public ghostBytes5;
+    bytes public ghostBytes6;
+    bytes public ghostBytes7;
+    bytes public ghostBytes8;
+    bytes public ghostBytes9;
+
+    function getGhostBytesArray() public view returns (bytes[] memory _arr) {
+        _arr = new bytes[](10);
+        _arr[0] = ghostBytes0;
+        _arr[1] = ghostBytes1;
+        _arr[2] = ghostBytes2;
+        _arr[3] = ghostBytes3;
+        _arr[4] = ghostBytes4;
+        _arr[5] = ghostBytes5;
+        _arr[6] = ghostBytes6;
+        _arr[7] = ghostBytes7;
+        _arr[8] = ghostBytes8;
+        _arr[9] = ghostBytes9;
+    }
+}
+
+/// @notice tests inheriting this contract cannot be run with forge
+abstract contract KontrolUtils is KontrolCheats {
+    /// @dev we only care about the vm signature
+    // Cheat code address, 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D.
+    address private constant VM_ADDRESS = address(uint160(uint256(keccak256("hevm cheat code"))));
+    Vm private constant vm = Vm(VM_ADDRESS);
+
+    function createWithdrawalTransaction(
+        uint256 _tx0,
+        address _tx1,
+        address _tx2,
+        uint256 _tx3,
+        uint256 _tx4,
+        bytes memory _tx5
+    )
+        internal
+        pure
+        returns (Types.WithdrawalTransaction memory _tx)
+    {
+        _tx = Types.WithdrawalTransaction(_tx0, _tx1, _tx2, _tx3, _tx4, _tx5);
+    }
+
+    function freshBytesArray(uint256 symbolicArrayLength) public returns (bytes[] memory symbolicArray) {
+        symbolicArray = new bytes[](symbolicArrayLength);
+
+        for (uint256 i = 0; i < symbolicArray.length; ++i) {
+            symbolicArray[i] = abi.encodePacked(kevm.freshUInt(32));
```

Comment:
Nit, but `abi.encode(kevm.freshUInt(32))` and `bytes.concat(bytes32(kevm.freshUInt(32)))` should result in the same value here. It's possible one of the other two is more performant in kontrol, though maybe a trivial difference? But if they all compile down to the same bytecode, which may be the case for this simple example, then it might not matter?